tests/end-to-end/e2e/viewer_test.go

```diff
@@ -0,0 +1,140 @@
+package e2e
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/kyma-incubator/compass/components/director/pkg/graphql"
+	"github.com/kyma-incubator/compass/tests/end-to-end/pkg/gql"
+	"github.com/kyma-incubator/compass/tests/end-to-end/pkg/idtokenprovider"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestViewerQuery(t *testing.T) {
+	domain := os.Getenv("DOMAIN")
+	require.NotEmpty(t, domain)
+	tenant := os.Getenv("DEFAULT_TENANT")
+	require.NotEmpty(t, tenant)
+	ctx := context.Background()
+
+	t.Log("Get Dex id_token")
+	config, err := idtokenprovider.NewConfigFromEnv()
+	require.NoError(t, err)
+	dexToken, err := idtokenprovider.Authenticate(config)
+	require.NoError(t, err)
+
+	dexGraphQLClient := gql.NewAuthorizedGraphQLClient(dexToken)
+
+	t.Run("Test viewer as Integration System", func(t *testing.T) {
+		t.Log("Create Integration System with Dex id token")
+		intSys := registerIntegrationSystem(t, ctx, dexGraphQLClient, tenant, "integration-system")
+
+		t.Logf("Registered Integration System with [id=%s]", intSys.ID)
+		defer unregisterIntegrationSystem(t, ctx, dexGraphQLClient, tenant, intSys.ID)
+
+		t.Log("Generate Client Credentials for Integration System")
+		intSysAuth := generateClientCredentialsForIntegrationSystem(t, ctx, dexGraphQLClient, tenant, intSys.ID)
+		intSysOauthCredentialData, ok := intSysAuth.Auth.Credential.(*graphql.OAuthCredentialData)
+		require.True(t, ok)
+		require.NotEmpty(t, intSysOauthCredentialData.ClientSecret)
+		require.NotEmpty(t, intSysOauthCredentialData.ClientID)
+
+		t.Log("Issue a Hydra token with Client Credentials")
```

Comment:
IMO we should not mention that this is token that comes from Hydra, this is implementation details, so it should be "get access token". Also, those lines:
```
t.Log("Issue a Hydra token with Client Credentials")
oauthCredentials := fmt.Sprintf("%s:%s", intSysOauthCredentialData.ClientID, intSysOauthCredentialData.ClientSecret)
		encodedCredentials := base64.StdEncoding.EncodeToString([]byte(oauthCredentials))
		hydraToken, err := fetchHydraAccessTokenWithScopesForApplication(t, encodedCredentials, intSysOauthCredentialData.URL)
		require.NoError(t, err)
```
can be replaced by calling one function: `getAccessToken`. 

Also, we can get Access token without any scopes, so we can have one func for Apps, Runtimes and Integration Sys